### PR TITLE
Backend app/feature/user

### PR DESCRIPTION
**Pull request by kiselev-danil**
Fix of user dto->model converter behavior, also fixed user update

### CODE REVIEW

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt [hunk 1]
```diff
@@ -87,5 +87,32 @@ class UserRepository {
...
+            throw UserNotFoundException("User ${model.fullName} with id:${model.id} does not exists")
+        }
+        var ent = db.users.find { it.id eq model.id!! }
+        ent?.tag = model.tag
+        ent?.fullName = model.fullName
+        ent?.active = model.active
+        ent?.avatarURL = model.avatarURL
+        ent?.role = model.role
+        ent?.flushChanges()
+        //todo integrations
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt:102]: We agreed that if we write a TODO, we mention who it applies to and leave a link to the board with the task.

Note: TODO(Egor Parkhomenko **`link`**) ...

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt [hunk 2]
```diff
@@ -87,5 +87,32 @@ class UserRepository {
         return UserTagModel(tag.id, tag.name)
     }
 
+    fun updateUser(model: UserModel): UserModel {
+        model.id ?: throw UserNotFoundException("No id in the model")
+        if (!existsById(model.id!!)) {
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt:92]: avoid !!

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt [hunk 3]
```diff
@@ -87,5 +87,32 @@ class UserRepository {
...
+        }
+        var ent = db.users.find { it.id eq model.id!! }
+        ent?.tag = model.tag
+        ent?.fullName = model.fullName
+        ent?.active = model.active
+        ent?.avatarURL = model.avatarURL
+        ent?.role = model.role
+        ent?.flushChanges()
+        //todo integrations
+        return findById(model.id!!)
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt:103]: avoid !!

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt [hunk 4]
```diff
@@ -87,5 +87,32 @@ class UserRepository {
         return UserTagModel(tag.id, tag.name)
     }
 
+    fun updateUser(model: UserModel): UserModel {
+        model.id ?: throw UserNotFoundException("No id in the model")
+        if (!existsById(model.id!!)) {
+            throw UserNotFoundException("User ${model.fullName} with id:${model.id} does not exists")
+        }
+        var ent = db.users.find { it.id eq model.id!! }
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt:95]: Why are you treating this variable as nullable if you did a check on the first line?

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/service/UserService.kt [hunk 5]
```diff
@@ -29,10 +29,9 @@ class UserService() : IUserService {
     }
 
     override fun updateUser(user: UserDTO, token: String): UserDTO {
-//        TODO fix update user inside repository
-//        val model = repository.updateUser(converterDTO.dTOToModel(user))
-//        return converterDTO.modelToDTO(model)
-        return user
+        val model = repository.updateUser(converterDTO.dTOToModel(user))
+        return converterDTO.modelToDTO(model)
+//        return user
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/service/UserService.kt:34]: Comments like this are best kept out of the pull request

**Review by 1MPULSEONE — CHANGES_REQUESTED** [review]
Fix problems

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt (hunk 2 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt:92]: Previous line checks if id is null

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/service/UserService.kt (hunk 5 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/service/UserService.kt]: Done

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt (hunk 4 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/repository/UserRepository.kt]: Solved

**Comment by 1MPULSEONE** [comment]
Btw fix commit style, check git flow guide

**Review by 1MPULSEONE — APPROVED** [review]

**Pull request merged**